File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.js
/**
 * @param {number[]} banned
 * @param {number} n
 * @param {number} maxSum
 * @return {number}
 */
var maxCount = function(banned, n, maxSum) {
    const st = new Set(banned);
    let arr = [];

    for(let i = 1; i <= n; i++){
        if (!st.has(i)){
            arr.push(i);
        }
    }

    let cursum = 0, count = 0;

    for (const it of arr){
        cursum += it;

        if (cursum <= maxSum){
            count += 1;
        }
        else{
            break;
        }
    }

    return count;
};

File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.py
class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        counter = Counter(banned)
        cursum = 0
        count = 0

        for i in range(1, n + 1):
            if i not in counter:
                cursum += i
                if cursum <= maxSum:
                    count += 1
                else:
                    break


        return count